src/Elmish/HTML/Events/Internal.purs

```diff
@@ -0,0 +1,66 @@
+-- | Internal implementation details. See comments on `Elmish.HTML.Events` instead.
+module Elmish.HTML.Events.Internal
+  ( REventBase
+  , RKeyboardEvent
+  , RModifierKeys
+  , RMouseEvent
+  , RSyntheticEvent
+  , RTouchEvent
+  )
+  where
+
+import Web.DOM (Element)
+import Web.Event.Internal.Types (Event)
+import Type.Row (type (+))
+
+type RSyntheticEvent = { | REventBase () }
+
+type RKeyboardEvent =
+  { charCode :: Int
+  , key :: String
```

Comment:
Do you think it’d be worth it to have an ADT for `key` at some point?

Reply:
Absolutely. At some point.